frontend/src/components/molecules/ColumnMenuFilter/index.jsx

```diff
@@ -32,32 +31,38 @@ const ColumnMenuFilter = (props) => {
     (state) => state[props.type]
   );
 
-  const onToggleClick = () => {
-    setIsOpen(!isOpen);
-  };
+  const onToggleClick = useCallback(() => {
+    setIsOpen((prev) => !prev);
+  }, []);
 
   const activeTableColumns = useMemo(
     () => tableColumns.map((col) => col.value),
     [tableColumns]
   );
 
-  const onSelect = (_event, value) => {
-    if (activeTableColumns.includes(value)) {
-      props.setColumns(value, false);
-    } else {
-      props.setColumns(value, true);
-    }
-  };
-  const toggle = (toggleRef) => (
-    <MenuToggle
-      ref={toggleRef}
-      variant="plain"
-      onClick={onToggleClick}
-      isExpanded={isOpen}
-      className="column-icon-menu"
-    >
-      <TableColumnIcon />
-    </MenuToggle>
+  const onSelect = useCallback(
+    (_event, value) => {
+      if (activeTableColumns.includes(value)) {
+        props.setColumns(value, false);
+      } else {
+        props.setColumns(value, true);
+      }
+    },
+    [activeTableColumns, props]
```

Comment:
So adding the dependencies fixes the problem? Wow. OK -- the mysteries of React and Javascript. But ... cool.